internal/wasi/wasi_test.go

```diff
@@ -1300,7 +1301,165 @@ func TestSnapshotPreview1_PathLink(t *testing.T) {
 	})
 }
 
-// TODO: TestSnapshotPreview1_PathOpen TestSnapshotPreview1_PathOpen_Errors
+func TestSnapshotPreview1_PathOpen(t *testing.T) {
+	fd := uint32(3)                              // arbitrary fd after 0, 1, and 2, that are stdin/out/err
+	dirflags := uint32(0)                        // arbitrary dirflags
+	path := uint32(1)                            // arbitrary offset
+	pathLen := uint32(6)                         // The length of path
+	oflags := uint32(0)                          // arbitrary oflags
+	fsRightsBase := uint64(wasi.R_FD_READ)       // arbitrary right
+	fsRightsInheriting := uint64(wasi.R_FD_READ) // arbitrary right
+	fdFlags := uint32(0)
+	resultOpenedFD := uint32(8)
+	initialMemory := []byte{
+		'?',                          // `path` is after this
+		'w', 'a', 'z', 'e', 'r', 'o', // path
+		'?', // `resultOpenedFD` is after this
+	}
+	expectedMemory := append(
+		initialMemory,
+		4, 0, 0, 0, // resultOpenedFD
+		'?',
+	)
+	expectedFd := uint32(4) // arbitrary expected FD
```

Comment:
I'm thinking whether we should consistently use `FD` or `Fd`. In general we should use `FD`, like `URL`  or `HTTP` as Golang community often follows https://github.com/golang/go/wiki/CodeReviewComments#initialisms. On the other hand, Golang's std library uses the word `Fd` https://pkg.go.dev/os#File.Fd. Plus considering that `FdClose` or other `FdSomething` looks more readable than `FDClose`, I'd like to say `Fd` is better. wdyt?

Reply:
for translated things, I think better to follow the delimiter rules as ex fd_close -> FdClose is more obvious and mechanical than FDClose. Also bear in mind these functions aren't public.

For things we define that aren't mapped to something external, probably we would tend to use Go rules which as you note aren't followed always. I often am in the same boat, where I'm looking to see how often someone does HTTPThing vs HttpThing and if it is a clear majority or not 😅

Reply:
one reason I've been using normal case format conversion for things from the spec as there is so much to translate and sometimes it is subtle, like something ending in _s. We end up being near code generators and to manage the scale of that is easier vs having to think if something is really an abbreviation or not. ex iovs? To not have to think carefully *1000 things is its own benefit and if someone gets really upset it would only be internal code so we can change that any time.

Basically we can save the higher scrutiny for code outside /internal as that's the stuff if you change breaks api (or is it API?) 😎

Reply:
Adopted "Fd" for semi-automatic conversions, and "FD" for normal Go names. Thanks!